src/Swarm/TUI/Controller.hs

```diff
@@ -223,7 +223,15 @@ handleMainEvent s = \case
   FKey 4 | not (null (s ^. gameState . availableCommands . notificationsContent)) -> do
     s' <- toggleModal s CommandsModal
     continue (s' & gameState . availableCommands . notificationsCount .~ 0)
-  FKey 5 -> toggleModal s MessagesModal >>= continue
+  FKey 5 -> do
+    let t = s ^. gameState . ticks
+    s' <- toggleModal s MessagesModal
+    liftIO $ appendFile "/tmp/debug.log" $ "\n\nOLD LEN - " <> show (s ^. gameState . messageNotifications . notificationsCount)
+    liftIO $ appendFile "/tmp/debug.log" $ "\nOLD TIME - " <> show (s' ^. gameState . lastSeenMessageTime)
+    liftIO $ appendFile "/tmp/debug.log" $ "\nNEW TIME - " <> show t
+    liftIO $ appendFile "/tmp/debug.log" $ "\nNEW LEN - " <> show (s' ^. gameState . messageNotifications . notificationsCount)
+    liftIO $ appendFile "/tmp/debug.log" $ "\nMESSAGES - " <> show (view leTime <$> s' ^. gameState . messageQueue)
+    continue (s' & gameState . lastSeenMessageTime .~ t)
```

Comment:
Somehow, I am not counting the number of new messages right and it's always zero! :facepalm: 

The flow in Controller should be correct, so it has to be some stupid mistake in State.

Reply:
Should work now! :heavy_check_mark: